Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,191 @@
+                                    pruning = TRUE){
+  ### total length
+  l <- length(target.mat[, 1])
+  
+  ### train and test index for each fold
+  fold_split_idx <- NULL
+  
+  ### initial row
+  fold_split_idx$train <- seq(from = 1, to = ((l/n_folds)*(n_folds-1)), by = 1)
+  fold_split_idx$test <- seq(from = ((l/n_folds)*(n_folds-1)) + 1, to = l, by = 1)
+  
+  ### middle values
+  for(n in (n_folds-1):2){
+    this_row_train <- c(seq(from = 1, to = ((l/n_folds)*(n-1)), by = 1), seq(from = as.integer((l/n_folds)*(n)+1), to = l, by = 1))
+    this_row_test <- seq(from = ((l/n_folds)*(n-1)) + 1, to = as.integer((l/n_folds)*(n)))
+    
+    fold_split_idx$train <- rbind(fold_split_idx$train, this_row_train)
+    fold_split_idx$test <- rbind(fold_split_idx$test, this_row_test)
+  }
+  
+  ### end row
+  this_row_train <- seq(from = (as.integer((l/n_folds)+1)), to = l, by = 1)
+  this_row_test <- seq(from = 1, to = as.integer((l/n_folds)), by = 1)
+  
+  fold_split_idx$train <- rbind(fold_split_idx$train, this_row_train)
+  fold_split_idx$test <- rbind(fold_split_idx$test, this_row_test)
+  
+  ### fold trees
+  fold_tree <- NULL
+  for(i in 1:n_folds){
+    fold_tree[[i]] <- mmit(target.mat[fold_split_idx$train[i,],], feature.mat[fold_split_idx$train[i,],],  
+                                       maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                                       loss = loss, min_sample = as.numeric(parameters$min_sample))
+  }
+  
+  ### master tree
+  master_tree <- mmit(target.mat, feature.mat,  
+                      maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                      loss = loss, min_sample = as.numeric(parameters$min_sample))
+  
+  
+  ### if pruning
+  if(pruning){
+    # Get the pruned master and cross-validation trees
+    master_data <- pruning(master_tree)
+    master_alphas <- master_data[, 1]
+    master_pruned_trees <- master_data[, -1]
+    
+    fold_alphas <- NULL
+    fold_prune_trees <- NULL
+    for(i in 1 : n_folds){
+      fold_data <- pruning(fold_tree[[i]])
+      fold_alphas[[i]] <- fold_data[, 1] 
+      fold_prune_trees[[i]] <- fold_data[, -1]
+    }
+    
+    ### alphas list should not contain repeating alpha
+    master_alphas <- master_alphas[!duplicated(master_alphas)]
+    fold_alphas <- fold_alphas[!duplicated(fold_alphas)]
+    
+    
+    # Compute the test risk for all pruned trees of each fold
+    alpha_path_score <- NULL
+    for(i in 1 : n_folds){
+      for(j in 1 : length(fold_prune_trees[[i]][,1])){
+        ### convert pruned tree list to partynode
+        node <- fold_prune_trees[[i]][[j]]
+        
+        fit <- fitted_node(node, feature.mat[fold_split_idx$test[i,],])
+        n <- nodeapply(node, ids = fit, info_node)
+        ###prediction of test data
+        prediction <- matrix(unlist(n), nrow = length(n), byrow = T)[,1]
```

Comment:
Why are the predictions extracted from info node? You should simply be applying the fitted mmit tree to the data to compute predictions. Doesn't partykit provide a method for computing predictions with a tree?

Reply:
Answered in live meeting. 👍